File: Difficulty: Easy/Height of Binary Tree/height-of-binary-tree.cpp
/*
class Node {
public:
    int data;
    Node* left;
    Node* right;

    Node(int val) {
        data = val;
        left = right = NULL;
    }
};
*/

class Solution {
  public:
    int height(Node* root) {
        // code here
        
        if(root->left == nullptr && root->right==nullptr) {
            return 0;
        }
        
        int left = 0;
        
        if(root->left)
            left = height(root->left);
        int right = 0;
        if(root->right)
            right = height(root->right);
        
        return max(left,right)+1;
    }
};

File: Difficulty: Easy/Height of Binary Tree/height-of-binary-tree.java
/*
Definition for Node
class Node {
    int data;
    Node left, right;

    Node(int val)
    {
        data = val;
        left = right = null;
    }
}
*/

class Solution {
    public int height(Node root) {
        // code here
        if(root==null) return -1;
        return Math.max(height(root.left),height(root.right))+1;
    }
}